### ISSUE TITLE
To make build cross-platform, devs should be able to perform Deploy builds on MS Windows

### ISSUE THREAD

**Issue by craigfowler** [csf-dev/ZPT-Sharp#132]
Now that #133 is resolved, build in the `Debug` or `Release` configurations work OK on MS Windows.

`Deploy`-mode builds have a few issues, some of which are listed below.  The scope of this ticket is currently incomplete though. Other problems need adding and fixing add they arise.

This ticket is complete when it is possible to perform a complete deployment on a Windows machine (assuming dependencies are present).
## Path environment variable

Using `cmd.exe`, in order to get the relevant commands into your path, you must first execute
a series of `SET PATH` commands to get the relevant apps into your path environment variable, such as:

```
SET PATH=%PATH%;c:\cygwin64\bin
```
## Man command fails

The export of the manual page to text (in the CLI project) doesn't work on Windows.  In order to fix that, the command needs to be as follows:

```
SET "MANWIDTH=80" && man --nh -l ZptBuilder.1 > bin\Release\Doc\ZptBuilder.MANUAL.txt
```

This will have to be performed using OS detection, and then using a different command for Windows.

The above command assumes #136 is fixed.

**Comment by craigfowler** [comment]
Additionally man2html is a dependency, which needs to be provided. It's not available in cygwin but perhaps is in MinGW.

**Comment by craigfowler** [comment]
Current issues building on Windows:

```
Microsoft (R) Build Engine version 14.0.25420.1
Copyright (C) Microsoft Corporation. All rights reserved.

Building the projects in this solution one at a time. To enable parallel build, please add the "/m" switch.
Build started 07/11/2016 17:08:39.
Project "C:\repos\ZPT-Sharp\CSF.Zpt.sln" on node 1 (default targets).
ValidateSolutionConfiguration:
  Building solution configuration "Debug|Any CPU".
Project "C:\repos\ZPT-Sharp\CSF.Zpt.sln" (1) is building "C:\repos\ZPT-Sharp\CSF.Zpt\CSF.Zpt.csproj" (2) on node 1 (default targets).
RestoreStubAssemblyInfo:
  Copying file from "..\Common\AssemblyInfo.stub.cs" to "AssemblyInfo.cs".
C:\repos\ZPT-Sharp\Deployment\CSF.Zpt.DeploymentTasks\CSF.Zpt.DeploymentTasks.targets(96,5): error MSB4062: The "ReadVersionNumber" task could not be loaded from the assembly C:\repos\ZPT-Sharp\Deployment\CSF.Zpt.DeploymentTasks\bin\Debug\CSF.Zpt.DeploymentTasks.dll. Could not load file or assembly 'file:///C:\repos\ZPT-Sharp\Deployment\CSF.Zpt.DeploymentTasks\bin\Debug\CSF.Zpt.DeploymentTasks.dll' or one of its dependencies. The system cannot find the file specified. Confirm that the <UsingTask> declaration is correct, that the assembly and all its dependencies are available, and that the task contains a public class that implements Microsoft.Build.Framework.ITask. [C:\repos\ZPT-Sharp\CSF.Zpt\CSF.Zpt.csproj]
Done Building Project "C:\repos\ZPT-Sharp\CSF.Zpt\CSF.Zpt.csproj" (default targets) -- FAILED.
Project "C:\repos\ZPT-Sharp\CSF.Zpt.sln" (1) is building "C:\repos\ZPT-Sharp\CSF\CSF\CSF.csproj" (3) on node 1 (default targets).
CoreResGen:
  No resources are out of date with respect to their source files. Skipping resource generation.
GenerateTargetFrameworkMonikerAttribute:
Skipping target "GenerateTargetFrameworkMonikerAttribute" because all output files are up-to-date with respect to the input files.
CoreCompile:
Skipping target "CoreCompile" because all output files are up-to-date with respect to the input files.
CopyFilesToOutputDirectory:
  CSF -> C:\repos\ZPT-Sharp\CSF\CSF\bin\Debug\CSF.dll
Done Building Project "C:\repos\ZPT-Sharp\CSF\CSF\CSF.csproj" (default targets).
Project "C:\repos\ZPT-Sharp\CSF.Zpt.sln" (1) is building "C:\repos\ZPT-Sharp\CSF\CSF.Cli.Parameters\CSF.Cli.Parameters.csproj" (4) on node 1 (default targets).
Project "C:\repos\ZPT-Sharp\CSF\CSF.Cli.Parameters\CSF.Cli.Parameters.csproj" (4) is building "C:\repos\ZPT-Sharp\CSF\CSF.Reflection\CSF.Reflection.csproj" (5) on node 1 (default targets).
CoreResGen:
  No resources are out of date with respect to their source files. Skipping resource generation.
GenerateTargetFrameworkMonikerAttribute:
Skipping target "GenerateTargetFrameworkMonikerAttribute" because all output files are up-to-date with respect to the input files.
CoreCompile:
Skipping target "CoreCompile" because all output files are up-to-date with respect to the input files.
CopyFilesToOutputDirectory:
  CSF.Reflection -> C:\repos\ZPT-Sharp\CSF\CSF.Reflection\bin\Debug\CSF.Reflection.dll
Done Building Project "C:\repos\ZPT-Sharp\CSF\CSF.Reflection\CSF.Reflection.csproj" (default targets).
CoreResGen:
  No resources are out of date with respect to their source files. Skipping resource generation.
GenerateTargetFrameworkMonikerAttribute:
Skipping target "GenerateTargetFrameworkMonikerAttribute" because all output files are up-to-date with respect to the input files.
CoreCompile:
Skipping target "CoreCompile" because all output files are up-to-date with respect to the input files.
CopyFilesToOutputDirectory:
  CSF.Cli.Parameters -> C:\repos\ZPT-Sharp\CSF\CSF.Cli.Parameters\bin\Debug\CSF.Cli.Parameters.dll
Done Building Project "C:\repos\ZPT-Sharp\CSF\CSF.Cli.Parameters\CSF.Cli.Parameters.csproj" (default targets).
Project "C:\repos\ZPT-Sharp\CSF.Zpt.sln" (1) is building "C:\repos\ZPT-Sharp\CSF\CSF.Configuration\CSF.Configuration.csproj" (6) on node 1 (default targets).
GenerateTargetFrameworkMonikerAttribute:
Skipping target "GenerateTargetFrameworkMonikerAttribute" because all output files are up-to-date with respect to the input files.
CoreCompile:
Skipping target "CoreCompile" because all output files are up-to-date with respect to the input files.
CopyFilesToOutputDirectory:
  CSF.Configuration -> C:\repos\ZPT-Sharp\CSF\CSF.Configuration\bin\Debug\CSF.Configuration.dll
Done Building Project "C:\repos\ZPT-Sharp\CSF\CSF.Configuration\CSF.Configuration.csproj" (default targets).
Project "C:\repos\ZPT-Sharp\CSF.Zpt.sln" (1) is building "C:\repos\ZPT-Sharp\CSF\CSF.IniDocuments\CSF.IniDocuments.csproj" (7) on node 1 (default targets).
GenerateTargetFrameworkMonikerAttribute:
Skipping target "GenerateTargetFrameworkMonikerAttribute" because all output files are up-to-date with respect to the input files.
CoreCompile:
Skipping target "CoreCompile" because all output files are up-to-date with respect to the input files.
CopyFilesToOutputDirectory:
  CSF.IniDocuments -> C:\repos\ZPT-Sharp\CSF\CSF.IniDocuments\bin\Debug\CSF.IniDocuments.dll
Done Building Project "C:\repos\ZPT-Sharp\CSF\CSF.IniDocuments\CSF.IniDocuments.csproj" (default targets).
Project "C:\repos\ZPT-Sharp\CSF.Zpt.sln" (1) is building "C:\repos\ZPT-Sharp\CSF.Zpt.MVC\CSF.Zpt.MVC.csproj" (8) on node 1 (default targets).
PrepareForBuild:
  Creating directory "bin\Debug\".
  Creating directory "obj\Debug\".
Project "C:\repos\ZPT-Sharp\CSF.Zpt.MVC\CSF.Zpt.MVC.csproj" (8) is building "C:\repos\ZPT-Sharp\Deployment\CSF.Zpt.DeploymentTasks\CSF.Zpt.DeploymentTasks.csproj" (9) on node 1 (default targets).
PrepareForBuild:
  Creating directory "bin\Debug\".
  Creating directory "obj\Debug\".
CoreResGen:
  "C:\Program Files (x86)\Microsoft SDKs\Windows\v10.0A\bin\NETFX 4.6 Tools\resgen.exe" /useSourcePath /r:C:\repos\ZPT-Sharp\CSF\CSF.IniDocuments\bin\Debug\CSF.IniDocuments.dll /r:"C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.5\Microsoft.Build.Framework.dll" /r:"C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.5\Microsoft.Build.Tasks.v4.0.dll" /r:"C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.5\Microsoft.Build.Utilities.v4.0.dll" /r:"C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.5\mscorlib.dll" /r:"C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.5\System.Core.dll" /r:"C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.5\System.dll" /r:"C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.5\System.IO.Compression.dll" /compile Resources\ErrorMessages.resx,obj\Debug\CSF.Zpt.DeploymentTasks.Resources.ErrorMessages.resources
  Processing resource file "Resources\ErrorMessages.resx" into "obj\Debug\CSF.Zpt.DeploymentTasks.Resources.ErrorMessages.resources".
GenerateTargetFrameworkMonikerAttribute:
Skipping target "GenerateTargetFrameworkMonikerAttribute" because all output files are up-to-date with respect to the input files.
CoreCompile:
  C:\Program Files (x86)\MSBuild\14.0\bin\csc.exe /noconfig /nowarn:1701,1702 /nostdlib+ /errorreport:prompt /warn:4 /define:DEBUG /highentropyva+ /reference:C:\repos\ZPT-Sharp\CSF\CSF.IniDocuments\bin\Debug\CSF.IniDocuments.dll /reference:"C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.5\Microsoft.Build.Framework.dll" /reference:"C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.5\Microsoft.Build.Tasks.v4.0.dll" /reference:"C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.5\Microsoft.Build.Utilities.v4.0.dll" /reference:"C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.5\mscorlib.dll" /reference:"C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.5\System.Core.dll" /reference:"C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.5\System.dll" /reference:"C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.5\System.IO.Compression.dll" /debug+ /debug:full /optimize- /out:obj\Debug\CSF.Zpt.DeploymentTasks.dll /ruleset:"C:\Program Files (x86)\Microsoft Visual Studio 14.0\Team Tools\Static Analysis Tools\\Rule Sets\MinimumRecommendedRules.ruleset" /subsystemversion:6.00 /resource:obj\Debug\CSF.Zpt.DeploymentTasks.Resources.ErrorMessages.resources /target:library /utf8output AssemblyInfo.cs ExecWithIgnoredExitCodes.cs FixDocTag.cs CreateZipArchive.cs Resources\ErrorMessages.Designer.cs RequireFilesOnPath.cs ReadVersionNumber.cs IncrementBuildVersion.cs "C:\cygwin64\tmp\.NETFramework,Version=v4.5.AssemblyAttributes.cs"
  Using shared compilation with compiler from directory: C:\Program Files (x86)\MSBuild\14.0\bin
_CopyFilesMarkedCopyLocal:
  Copying file from "C:\repos\ZPT-Sharp\CSF\CSF.IniDocuments\bin\Debug\CSF.IniDocuments.dll" to "bin\Debug\CSF.IniDocuments.dll".
  Copying file from "C:\repos\ZPT-Sharp\CSF\CSF.IniDocuments\bin\Debug\CSF.IniDocuments.pdb" to "bin\Debug\CSF.IniDocuments.pdb".
_CopyOutOfDateSourceItemsToOutputDirectoryAlways:
  Copying file from "C:\repos\ZPT-Sharp\Deployment\CSF.Zpt.DeploymentTasks\CSF.Zpt.DeploymentTasks.targets" to "bin\Debug\CSF.Zpt.DeploymentTasks.targets".
CopyFilesToOutputDirectory:
  Copying file from "obj\Debug\CSF.Zpt.DeploymentTasks.dll" to "bin\Debug\CSF.Zpt.DeploymentTasks.dll".
  CSF.Zpt.DeploymentTasks -> C:\repos\ZPT-Sharp\Deployment\CSF.Zpt.DeploymentTasks\bin\Debug\CSF.Zpt.DeploymentTasks.dll
  Copying file from "obj\Debug\CSF.Zpt.DeploymentTasks.pdb" to "bin\Debug\CSF.Zpt.DeploymentTasks.pdb".
Done Building Project "C:\repos\ZPT-Sharp\Deployment\CSF.Zpt.DeploymentTasks\CSF.Zpt.DeploymentTasks.csproj" (default targets).
Project "C:\repos\ZPT-Sharp\CSF.Zpt.MVC\CSF.Zpt.MVC.csproj" (8) is building "C:\repos\ZPT-Sharp\CSF.Zpt.Abstractions\CSF.Zpt.Abstractions.csproj" (10) on node 1 (default targets).
RestoreStubAssemblyInfo:
  Copying file from "..\Common\AssemblyInfo.stub.cs" to "AssemblyInfo.cs".
C:\repos\ZPT-Sharp\Deployment\CSF.Zpt.DeploymentTasks\CSF.Zpt.DeploymentTasks.targets(96,5): error MSB4062: The "ReadVersionNumber" task could not be loaded from the assembly C:\repos\ZPT-Sharp\Deployment\CSF.Zpt.DeploymentTasks\bin\Debug\CSF.Zpt.DeploymentTasks.dll. Could not load file or assembly 'file:///C:\repos\ZPT-Sharp\Deployment\CSF.Zpt.DeploymentTasks\bin\Debug\CSF.Zpt.DeploymentTasks.dll' or one of its dependencies. The system cannot find the file specified. Confirm that the <UsingTask> declaration is correct, that the assembly and all its dependencies are available, and that the task contains a public class that implements Microsoft.Build.Framework.ITask. [C:\repos\ZPT-Sharp\CSF.Zpt.Abstractions\CSF.Zpt.Abstractions.csproj]
Done Building Project "C:\repos\ZPT-Sharp\CSF.Zpt.Abstractions\CSF.Zpt.Abstractions.csproj" (default targets) -- FAILED.
Project "C:\repos\ZPT-Sharp\CSF.Zpt.MVC\CSF.Zpt.MVC.csproj" (8) is building "C:\repos\ZPT-Sharp\CSF.Zpt.Log4net\CSF.Zpt.Log4net.csproj" (11) on node 1 (default targets).
PrepareForBuild:
  Creating directory "bin\Debug\".
  Creating directory "obj\Debug\".
Done Building Project "C:\repos\ZPT-Sharp\CSF.Zpt.Log4net\CSF.Zpt.Log4net.csproj" (default targets) -- FAILED.
Project "C:\repos\ZPT-Sharp\CSF.Zpt.MVC\CSF.Zpt.MVC.csproj" (8) is building "C:\repos\ZPT-Sharp\ExpressionEvaluators\CSF.Zpt.ExpressionEvaluators.NotExpressions\CSF.Zpt.ExpressionEvaluators.NotExpressions.csproj" (12) on node 1 (default targets).
PrepareForBuild:
  Creating directory "bin\Debug\".
  Creating directory "obj\Debug\".
Done Building Project "C:\repos\ZPT-Sharp\ExpressionEvaluators\CSF.Zpt.ExpressionEvaluators.NotExpressions\CSF.Zpt.ExpressionEvaluators.NotExpressions.csproj" (default targets) -- FAILED.
Project "C:\repos\ZPT-Sharp\CSF.Zpt.MVC\CSF.Zpt.MVC.csproj" (8) is building "C:\repos\ZPT-Sharp\ExpressionEvaluators\CSF.Zpt.ExpressionEvaluators.PathExpressions\CSF.Zpt.ExpressionEvaluators.PathExpressions.csproj" (13) on node 1 (default targets).
PrepareForBuild:
  Creating directory "bin\Debug\".
  Creating directory "obj\Debug\".
Done Building Project "C:\repos\ZPT-Sharp\ExpressionEvaluators\CSF.Zpt.ExpressionEvaluators.PathExpressions\CSF.Zpt.ExpressionEvaluators.PathExpressions.csproj" (default targets) -- FAILED.
Project "C:\repos\ZPT-Sharp\CSF.Zpt.MVC\CSF.Zpt.MVC.csproj" (8) is building "C:\repos\ZPT-Sharp\ExpressionEvaluators\CSF.Zpt.ExpressionEvaluators.StringExpressions\CSF.Zpt.ExpressionEvaluators.StringExpressions.csproj" (14) on node 1 (default targets).
PrepareForBuild:
  Creating directory "bin\Debug\".
  Creating directory "obj\Debug\".
Done Building Project "C:\repos\ZPT-Sharp\ExpressionEvaluators\CSF.Zpt.ExpressionEvaluators.StringExpressions\CSF.Zpt.ExpressionEvaluators.StringExpressions.csproj" (default targets) -- FAILED.
Project "C:\repos\ZPT-Sharp\CSF.Zpt.MVC\CSF.Zpt.MVC.csproj" (8) is building "C:\repos\ZPT-Sharp\DocumentProviders\CSF.Zpt.DocumentProviders.HtmlHAP\CSF.Zpt.DocumentProviders.HtmlHAP.csproj" (15) on node 1 (default targets).
PrepareForBuild:
  Creating directory "bin\Debug\".
  Creating directory "obj\Debug\".
Done Building Project "C:\repos\ZPT-Sharp\DocumentProviders\CSF.Zpt.DocumentProviders.HtmlHAP\CSF.Zpt.DocumentProviders.HtmlHAP.csproj" (default targets) -- FAILED.
Project "C:\repos\ZPT-Sharp\CSF.Zpt.MVC\CSF.Zpt.MVC.csproj" (8) is building "C:\repos\ZPT-Sharp\DocumentProviders\CSF.Zpt.DocumentProviders.XmlLinq\CSF.Zpt.DocumentProviders.XmlLinq.csproj" (16) on node 1 (default targets).
PrepareForBuild:
  Creating directory "bin\Debug\".
  Creating directory "obj\Debug\".
Done Building Project "C:\repos\ZPT-Sharp\DocumentProviders\CSF.Zpt.DocumentProviders.XmlLinq\CSF.Zpt.DocumentProviders.XmlLinq.csproj" (default targets) -- FAILED.
Done Building Project "C:\repos\ZPT-Sharp\CSF.Zpt.MVC\CSF.Zpt.MVC.csproj" (default targets) -- FAILED.
Project "C:\repos\ZPT-Sharp\CSF.Zpt.sln" (1) is building "C:\repos\ZPT-Sharp\CSF.Zpt.Cli\CSF.Zpt.Cli.csproj" (17) on node 1 (default targets).
PrepareForBuild:
  Creating directory "bin\Debug\".
  Creating directory "obj\Debug\".
Done Building Project "C:\repos\ZPT-Sharp\CSF.Zpt.Cli\CSF.Zpt.Cli.csproj" (default targets) -- FAILED.
Project "C:\repos\ZPT-Sharp\CSF.Zpt.sln" (1) is building "C:\repos\ZPT-Sharp\DocumentProviders\CSF.Zpt.DocumentProviders.XmlLegacy\CSF.Zpt.DocumentProviders.XmlLegacy.csproj" (18) on node 1 (default targets).
PrepareForBuild:
  Creating directory "bin\Debug\".
  Creating directory "obj\Debug\".
Done Building Project "C:\repos\ZPT-Sharp\DocumentProviders\CSF.Zpt.DocumentProviders.XmlLegacy\CSF.Zpt.DocumentProviders.XmlLegacy.csproj" (default targets) -- FAILED.
Project "C:\repos\ZPT-Sharp\CSF.Zpt.sln" (1) is building "C:\repos\ZPT-Sharp\Tests\Test.CSF.Zpt\Test.CSF.Zpt.csproj" (19) on node 1 (default targets).
PrepareForBuild:
  Creating directory "bin\Debug\".
  Creating directory "obj\Debug\".
Project "C:\repos\ZPT-Sharp\Tests\Test.CSF.Zpt\Test.CSF.Zpt.csproj" (19) is building "C:\repos\ZPT-Sharp\Tests\CSF.Zpt.TestUtils\CSF.Zpt.TestUtils.csproj" (20) on node 1 (default targets).
PrepareForBuild:
  Creating directory "bin\Debug\".
  Creating directory "obj\Debug\".
Done Building Project "C:\repos\ZPT-Sharp\Tests\CSF.Zpt.TestUtils\CSF.Zpt.TestUtils.csproj" (default targets) -- FAILED.
Done Building Project "C:\repos\ZPT-Sharp\Tests\Test.CSF.Zpt\Test.CSF.Zpt.csproj" (default targets) -- FAILED.
Project "C:\repos\ZPT-Sharp\CSF.Zpt.sln" (1) is building "C:\repos\ZPT-Sharp\Tests\Test.CSF.Zpt.Cli\Test.CSF.Zpt.Cli.csproj" (21) on node 1 (default targets).
PrepareForBuild:
  Creating directory "bin\Debug\".
  Creating directory "obj\Debug\".
Done Building Project "C:\repos\ZPT-Sharp\Tests\Test.CSF.Zpt.Cli\Test.CSF.Zpt.Cli.csproj" (default targets) -- FAILED.
Project "C:\repos\ZPT-Sharp\CSF.Zpt.sln" (1) is building "C:\repos\ZPT-Sharp\Tests\Test.CSF.Zpt.Abstractions\Test.CSF.Zpt.Abstractions.csproj" (22) on node 1 (default targets).
PrepareForBuild:
  Creating directory "bin\Debug\".
  Creating directory "obj\Debug\".
Done Building Project "C:\repos\ZPT-Sharp\Tests\Test.CSF.Zpt.Abstractions\Test.CSF.Zpt.Abstractions.csproj" (default targets) -- FAILED.
Project "C:\repos\ZPT-Sharp\CSF.Zpt.sln" (1) is building "C:\repos\ZPT-Sharp\Tests\Test.CSF.Zpt.DocumentProviders.HtmlHAP\Test.CSF.Zpt.DocumentProviders.HtmlHAP.csproj" (23) on node 1 (default targets).
PrepareForBuild:
  Creating directory "bin\Debug\".
  Creating directory "obj\Debug\".
Done Building Project "C:\repos\ZPT-Sharp\Tests\Test.CSF.Zpt.DocumentProviders.HtmlHAP\Test.CSF.Zpt.DocumentProviders.HtmlHAP.csproj" (default targets) -- FAILED.
Project "C:\repos\ZPT-Sharp\CSF.Zpt.sln" (1) is building "C:\repos\ZPT-Sharp\Tests\Test.CSF.Zpt.DocumentProviders.XmlLegacy\Test.CSF.Zpt.DocumentProviders.XmlLegacy.csproj" (24) on node 1 (default targets).
PrepareForBuild:
  Creating directory "bin\Debug\".
  Creating directory "obj\Debug\".
Done Building Project "C:\repos\ZPT-Sharp\Tests\Test.CSF.Zpt.DocumentProviders.XmlLegacy\Test.CSF.Zpt.DocumentProviders.XmlLegacy.csproj" (default targets) -- FAILED.
Project "C:\repos\ZPT-Sharp\CSF.Zpt.sln" (1) is building "C:\repos\ZPT-Sharp\Tests\Test.CSF.Zpt.Integration\Test.CSF.Zpt.Integration.csproj" (25) on node 1 (default targets).
PrepareForBuild:
  Creating directory "bin\Debug\".
  Creating directory "obj\Debug\".
Done Building Project "C:\repos\ZPT-Sharp\Tests\Test.CSF.Zpt.Integration\Test.CSF.Zpt.Integration.csproj" (default targets) -- FAILED.
Project "C:\repos\ZPT-Sharp\CSF.Zpt.sln" (1) is building "C:\repos\ZPT-Sharp\Tests\Test.CSF.Zpt.ExpressionEvaluators.NotExpressions\Test.CSF.Zpt.ExpressionEvaluators.NotExpressions.csproj" (26) on node 1 (default targets).
PrepareForBuild:
  Creating directory "bin\Debug\".
  Creating directory "obj\Debug\".
Done Building Project "C:\repos\ZPT-Sharp\Tests\Test.CSF.Zpt.ExpressionEvaluators.NotExpressions\Test.CSF.Zpt.ExpressionEvaluators.NotExpressions.csproj" (default targets) -- FAILED.
Project "C:\repos\ZPT-Sharp\CSF.Zpt.sln" (1) is building "C:\repos\ZPT-Sharp\Tests\Test.CSF.Zpt.ExpressionEvaluators.StringExpressions\Test.CSF.Zpt.ExpressionEvaluators.StringExpressions.csproj" (27) on node 1 (default targets).
PrepareForBuild:
  Creating directory "bin\Debug\".
  Creating directory "obj\Debug\".
Done Building Project "C:\repos\ZPT-Sharp\Tests\Test.CSF.Zpt.ExpressionEvaluators.StringExpressions\Test.CSF.Zpt.ExpressionEvaluators.StringExpressions.csproj" (default targets) -- FAILED.
Project "C:\repos\ZPT-Sharp\CSF.Zpt.sln" (1) is building "C:\repos\ZPT-Sharp\Tests\Test.CSF.Zpt.ExpressionEvaluators.PathExpressions\Test.CSF.Zpt.ExpressionEvaluators.PathExpressions.csproj" (28) on node 1 (default targets).
PrepareForBuild:
  Creating directory "bin\Debug\".
  Creating directory "obj\Debug\".
Done Building Project "C:\repos\ZPT-Sharp\Tests\Test.CSF.Zpt.ExpressionEvaluators.PathExpressions\Test.CSF.Zpt.ExpressionEvaluators.PathExpressions.csproj" (default targets) -- FAILED.
Done Building Project "C:\repos\ZPT-Sharp\CSF.Zpt.sln" (default targets) -- FAILED.

Build FAILED.

"C:\repos\ZPT-Sharp\CSF.Zpt.sln" (default target) (1) ->
"C:\repos\ZPT-Sharp\CSF.Zpt\CSF.Zpt.csproj" (default target) (2) ->
(StampAssemblyInfo target) -> 
  C:\repos\ZPT-Sharp\Deployment\CSF.Zpt.DeploymentTasks\CSF.Zpt.DeploymentTasks.targets(96,5): error MSB4062: The "ReadVersionNumber" task could not be loaded from the assembly C:\repos\ZPT-Sharp\Deployment\CSF.Zpt.DeploymentTasks\bin\Debug\CSF.Zpt.DeploymentTasks.dll. Could not load file or assembly 'file:///C:\repos\ZPT-Sharp\Deployment\CSF.Zpt.DeploymentTasks\bin\Debug\CSF.Zpt.DeploymentTasks.dll' or one of its dependencies. The system cannot find the file specified. Confirm that the <UsingTask> declaration is correct, that the assembly and all its dependencies are available, and that the task contains a public class that implements Microsoft.Build.Framework.ITask. [C:\repos\ZPT-Sharp\CSF.Zpt\CSF.Zpt.csproj]


"C:\repos\ZPT-Sharp\CSF.Zpt.sln" (default target) (1) ->
"C:\repos\ZPT-Sharp\CSF.Zpt.MVC\CSF.Zpt.MVC.csproj" (default target) (8) ->
"C:\repos\ZPT-Sharp\CSF.Zpt.Abstractions\CSF.Zpt.Abstractions.csproj" (default target) (10) ->
  C:\repos\ZPT-Sharp\Deployment\CSF.Zpt.DeploymentTasks\CSF.Zpt.DeploymentTasks.targets(96,5): error MSB4062: The "ReadVersionNumber" task could not be loaded from the assembly C:\repos\ZPT-Sharp\Deployment\CSF.Zpt.DeploymentTasks\bin\Debug\CSF.Zpt.DeploymentTasks.dll. Could not load file or assembly 'file:///C:\repos\ZPT-Sharp\Deployment\CSF.Zpt.DeploymentTasks\bin\Debug\CSF.Zpt.DeploymentTasks.dll' or one of its dependencies. The system cannot find the file specified. Confirm that the <UsingTask> declaration is correct, that the assembly and all its dependencies are available, and that the task contains a public class that implements Microsoft.Build.Framework.ITask. [C:\repos\ZPT-Sharp\CSF.Zpt.Abstractions\CSF.Zpt.Abstractions.csproj]

    0 Warning(s)
    2 Error(s)

Time Elapsed 00:00:01.15
```

**Comment by craigfowler** [comment]
The above issue is now fixed with explicit dependencies.  However this only proves that Debug/Release builds work and not Deploy.

**Comment by craigfowler** [comment]
Blocked by #169

**Comment by craigfowler** [comment]
As of `1a894601c12f4b5275f18aca50de12613b08da77` both Debug and Release builds work on Windows.

Debug builds pass all current tests.

Release builds work with two compile warnings from CSF core utils.

**Comment by craigfowler** [comment]
This is no longer relevant, the build process is greatly simplified now.
